src/widgets/splits/SplitContainer.cpp

```diff
@@ -799,6 +804,54 @@ void SplitContainer::popup()
     window.show();
 }
 
+NodeDescriptor SplitContainer::buildDescriptorRecursively(Node *currentNode)
+{
+    if (currentNode->children_.empty())
+    {
+        const auto channelTypeToString = [](Channel::Type type) {
+            switch (type)
+            {
+                case Channel::Type::Twitch:
+                    return "twitch";
+                case Channel::Type::TwitchWhispers:
+                    return "whispers";
+                case Channel::Type::TwitchWatching:
+                    return "watching";
+                case Channel::Type::TwitchMentions:
+                    return "mentions";
+                case Channel::Type::TwitchLive:
+                    return "live";
+                case Channel::Type::TwitchAutomod:
+                    return "automod";
+                case Channel::Type::Irc:
+                    return "irc";
+                case Channel::Type::Misc:
+                    return "misc";
+                default:
+                    return "";
+            }
+        };
+
+        SplitNodeDescriptor result;
+        result.type_ =
+            channelTypeToString(currentNode->split_->getChannel()->getType());
```

Comment:
@pajlada @Nerixyz not sure why but there is a `Channel::Type` inside `IndirectChannel` and inside `Channel`, and they are not matching here

![image](https://github.com/Chatterino/chatterino2/assets/43550602/0f427210-0c85-4bfd-af38-87ca7e8fe7b8)

The reason being that when the `watchingChannel` is created, it gets passed an empty channel that has type `None`

```
watchingChannel(Channel::getEmpty(), Channel::Type::TwitchWatching)
```

Reply:
The type of the IndirectChannel is the type of indirection. It's really only for the watching channel. Its type is only ever `TwitchWatching` or `Direct` (don't quote me on that). Maybe there should be another type for the type of indirection, but it's easier for saving/restoring right now, I suppose.